tests/multi_objective/test_trial.py

```diff
@@ -13,28 +14,33 @@
 def test_suggest() -> None:
     study = optuna.multi_objective.create_study(["maximize", "maximize"])
 
-    def objective(trial: optuna.multi_objective.trial.MultiObjectiveTrial) -> List[float]:
+    def objective(trial: optuna.multi_objective.trial.MultiObjectiveTrial) -> Tuple[float, float]:
         p0 = trial.suggest_float("p0", -10, 10)
         p1 = trial.suggest_uniform("p1", 3, 5)
         p2 = trial.suggest_loguniform("p2", 0.00001, 0.1)
         p3 = trial.suggest_discrete_uniform("p3", 100, 200, q=5)
         p4 = trial.suggest_int("p4", -20, -15)
-        p5 = trial.suggest_categorical("p5", [7, 1, 100])
+        p5 = cast(int, trial.suggest_categorical("p5", [7, 1, 100]))
```

Comment:
https://stackoverflow.com/questions/44739764/how-to-cast-a-typing-union-to-one-of-its-subtypes-in-python

Reply:
This is just an idea. It'd be nice if we could infer the return type of `suggest_categorical` in a generic way like "return int if int list given" instead of returning `Union` of python primitive types because this enforces users to call `cast` to a suggested value to resolve type errors when they use Optuna with a type-checking tool like `mypy`.